install/installer/README.md

```diff
@@ -13,3 +13,38 @@ The best way to get started with Gitpod is by using our recommended & default in
 > The installer is an internal tool and as such not expected to be used by those external to Gitpod. Instructions for how to use it can be found in the [docs](https://github.com/gitpod-io/gitpod/tree/main/install/installer/docs/overview.md) folder of this repo.
 
 [![Gitpod ready-to-code](https://img.shields.io/badge/Gitpod-ready--to--code-908a85?logo=gitpod)](https://gitpod.io/from-referrer/)
+
+## Configuration expectations
+
+The [config](./pkg/config/) is the primary way that the installer can be interacted with. This guide details the expectations that users can have of the installer's configuration and the contract that it specifies.
+
+### Versioning
+
+The installer supports only one active version of the configuration at a given time.
+
+#### Expectations for a version
+
+The configuration for a version has the following expectations. If a constraint needs to be broken to support a new feature, this should trigger a new version of the configuration surface.
```

Comment:
So, the reason for the opinion is largely to cover the use-case in the PAT secret - it's sort of back-filled. I would ordinarily agree that a change that causes an existing installation to fail is breaking.

There is another PR which also, strictly speaking, has introduced a breaking change (https://github.com/gitpod-io/gitpod/pull/14970#issuecomment-1344202161) - however, this is a change that no one is really using.

My actual (controversial) opinion is that we're actually using `v1` as a `v0` and that we can offer no real guarantees because of how much we're actually all changing it. Arguably, it makes no sense having guarantees until `v2`.

Do you think we're in a position to introduce a `v2` now?